javascript/javascript2/week3/Warmup.js

```diff
@@ -0,0 +1,142 @@
+//Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+
+function afterTheScriptIsLoaded(){
+  //console.log('Called after 2.5 seconds');
+  setTimeout(()=>console.log('Called after 5 seconds'),5000);
+}
+
+
+
+//Create a function that takes 2 parameters: delay and stringToLog. Calling this function should log out the stringToLog after delay seconds.
+//Call the function you have created with some different arguments.
+
+delayForSeconds(1,'delay string for 1 seconds');
+delayForSeconds(10,'delay string for 10 seconds');
+
+function delayForSeconds(delay,stringToLog){
+    setTimeout(()=>console.log(stringToLog),delay*1000);
+  }
+
+//Create a button in html.
+//When clicking this button, use the function you created in the previous task to
+ //log out the text: Called after 5 seconds 5 seconds after the button is clicked.
+ const clickButton = document.getElementById('button-one');
+ clickButton.addEventListener('click', afterTheScriptIsLoaded);
+
+
+//4th ecercise
+const logsOutEarth = function earth(){
+  console.log('Earth');
+};
+const logsOutSaturn = function saturn(){
+  console.log('Saturn');
+};
+function planetLogFunction(callback){
+    callback(planetLogFunction);
+}
+planetLogFunction(logsOutEarth);
+planetLogFunction(logsOutSaturn);
+
+
+//Create a button with the text called "Log location".
+//When this button is clicked the location (latitude, longitude) of the user should be
+//logged out using this browser api
+
+const logLocation =document.getElementById('button-two');
+logLocation.addEventListener('click', initMap);
+let map;
+//initMap();
+function initMap() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function showPosition(position) {
+      document.getElementById('co-ordinates').innerHTML = "Latitude: " + position.coords.latitude +
+      "<br>Longitude: " + position.coords.longitude;
+     map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: position.coords.latitude, lng: position.coords.longitude},
+        zoom: 8,
+      })
+    });
+    console.log(map);
+  }
+}
+
+
+//Create a function called runAfterDelay. It has two parameters: delay and callback. When called the function should wait delay seconds and then call the provided callback function.
+//Try and call this function with different delays and different callback functions
+
+function runAfterDelay(delay,callback){
+setTimeout(callback,delay*1000);
+}
+runAfterDelay(10,logsOutEarth/*function(){console.log('should logged after 5 seconds')}*/);
+
+//Check if we have double clicked on the page. A double click is defined by two clicks within 0.5 seconds.
+//If a double click has been detected, log out the text: "double click!"
+const functionDoubleClick = ()=>{
+  console.log('double click!');
+    //setTimeout(()=>{console.log('double click!')},0.5 *1000);
+  }
+document.getElementById('button-three').addEventListener('dblclick', functionDoubleClick);
+
```

Comment:
Its easy to log out the text: "double click!" using `dblclick `event. Try to rewrite using `click `event and `setTimeout.` It definitely takes little more efforts:)

Reply:
Thank you for your comments :).Sorry for late response, I will update now.